### DISCUSSION TITLE
act() and wait() methods

### DISCUSSION THREAD

**Discussion by StefanZivkovic** [testing-library/jest-dom#332] [Q&A]
```
await act(async () => {
      fireEvent.blur(state);
});
await wait(() => {
   expect(getByText('state is a required field')).toBeInTheDocument();
});
```

If we use act() around actions we simulate in tests, do we have to use wait() for incoming assertations?

**Comment by gnapse** [comment]
It depends, if the effect in the DOM resulting from the simulated event is not done synchronously by your components, then yes.
